app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_posts_counter
+    update(posts_counter: comments.count + likes.count)
+  end
```

Comment:
- [ ] Dear @ab-noori, You have done well adding most of the requirements in your app. But for the `update_post_counter` method we need to increment the post counter of the user who owns `this` post. Hence please update your method in a way it adds one to the post_counter of the author. 

> Something like...
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```

- [Optional] Also, it would be wise to set the `update_post_counter` to work alongside saving the new post. You cam make use of the `after_save :update_post_counter` helper to call the method every time a new post is saved.